Improve back navigation logic in ItemComponent

amadulhaxxani opened the pull request:
This pull request updates the navigation logic in the `ItemComponent` to improve how the back button behaves and how previous routes are handled. The main changes focus on making navigation more robust when returning to previous pages.

Navigation improvements:

* Updated the navigation logic in the `onBack` method to check if the previous URL matches the expected route; if so, it navigates to that URL, otherwise it uses `window.history.back()` to return to the last page.
* Modified the observable in `ngOnInit` to show the back button when the previous URL matches the expected route or is empty, making the UI more consistent for edge cases.

amadulhaxxani committed:
Improve back navigation logic in ItemComponent

Enhances the navigation logic to use window.history.back() when the previous URL does not match the expected route pattern. Also updates the back button visibility filter to allow empty URLs.

amadulhaxxani committed:
lint fix

amadulhaxxani committed:
copilot fixes

copilot fixes

amadulhaxxani committed:
Fix back button visibility logic in ItemComponent

Updated the showBackButton observable to correctly determine when the back button should be shown based on the previous route. Also added ngOnInit call in the related test to ensure proper initialization.

amadulhaxxani committed:
Remove unused 'filter' import from item.component.ts

The 'filter' operator from 'rxjs/operators' was imported but not used in item.component.ts. This commit cleans up the import statements.

kosarko commented on src/app/item-page/simple/item-types/shared/item.component.ts:
```diff
@@ -72,19 +72,21 @@ export class ItemComponent implements OnInit {
+            if (url && this.previousRoute.test(url)) {
+              this.router.navigateByUrl(url);
+            } else {
+              window.history.back();
```
I might have this wrong, but does this mean we always show a back button? I guess the original intention was to only show back button when you've moved to item view page from some other portion of the repository, e.g. if you come from search this takes you back to search.

I guess this also renders a back button if you come directly to item page (from email link, or just copying and pasting a handle), is it like that?

amadulhaxxani replied:
You are probably right about this
I will try to test and will fix it.

amadulhaxxani replied:
I am adding the session storage as it seems safe in our use case
accessing the routes from browser history seems to be not possible due to security reasons.
we will just store the key and use the lastUrl and I think it's working fine in our case now :) 
also when the url is directly accessed we are not showing this back button

amadulhaxxani committed:
Add session storage for previous URL in item page

Introduces generic methods in RouteService to store and retrieve URLs in session storage. ItemComponent now uses session storage to persist and retrieve the previous URL for improved back navigation, falling back to browser history if no valid URL is found.

amadulhaxxani committed:
Add session URL helpers to RouteService stubs in tests

Extended RouteService stubs in test files to include storeUrlInSession and getUrlFromSession methods, matching the interface used by ItemComponent and related tests. This ensures test mocks are up-to-date with the service's API.

amadulhaxxani committed:
Simplify back navigation logic in ItemComponent

Removed conditional check for previous URL and always navigate using router.navigateByUrl with storedPreviousUrl.
